File: app/src/main/java/com/karla/learningverbs/viewmodel/SignUpViewModel.java
package com.karla.learningverbs.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.karla.learningverbs.repository.userrepository.UserRepository2;
import com.karla.learningverbs.utils.firebase.CustomOnCompleteListener;
import com.karla.learningverbs.kotlin.utils.base.BaseViewModel;
import com.google.firebase.auth.AuthResult;

public class SignUpViewModel extends BaseViewModel {
    private MutableLiveData<Boolean> isSignUp;
    private UserRepository2 repository;

    public SignUpViewModel() {
        this.repository = UserRepository2.getInstance();
        repository = UserRepository2.getInstance();
        this.isSignUp = new MutableLiveData<>();
    }

    public void singUp(String email, String password) {
        loading.postValue(true);

        repository.singUp(new CustomOnCompleteListener<AuthResult>() {
            @Override
            public void onSuccess(AuthResult task) {
                isSignUp.postValue(true);
            }

            @Override
            public void onFailure(Throwable throwable) {
                msgError.postValue(throwable.getMessage());
            }

            @Override
            public void showLoaging() {
                loading.postValue(true);

            }

            @Override
            public void hideLoading() {
                loading.postValue(false);

            }
        }, email, password);
    }

    public LiveData<Boolean> getfirebaseUser() {
        return isSignUp;
    }

}

File: app/src/main/java/com/karla/learningverbs/kotlin/viewmodel/SignUpViewModel.kt
package com.karla.learningverbs.kotlin.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.google.firebase.auth.AuthResult
import com.karla.learningverbs.kotlin.repository.userrepository.UserRepository
import com.karla.learningverbs.kotlin.utils.base.BaseViewModel
import com.karla.learningverbs.kotlin.utils.firebase.CustomOnCompleteListener

class SignUpViewModel: BaseViewModel()  {

    private val isSignUp = MutableLiveData<Boolean>()
    private val repository: UserRepository = UserRepository()


    fun singUp(email: String, password: String) {
        loading.postValue(true)
        repository.singUp(object : CustomOnCompleteListener<AuthResult?>() {

            override fun onSuccess(task: AuthResult?) {
                isSignUp.postValue(true)
            }

            override fun onFailure(throwable: Throwable?) {
                throwable?.let {
                    msgError.postValue(it.message)
                }

            }

            override fun showLoaging() {
                loading.postValue(true)
            }

            override fun hideLoading() {
                loading.postValue(false)
            }
        }, email, password)
    }

    fun getfirebaseUser(): LiveData<Boolean?> {
        return isSignUp
    }
}